actor-typed-tests/src/test/scala/org/apache/pekko/actor/typed/DeadLetterSpec.scala

```diff
@@ -0,0 +1,135 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.pekko.actor.typed
+
+import org.apache.pekko.actor.IllegalActorStateException
+import org.apache.pekko.actor.testkit.typed.scaladsl.LogCapturing
+import org.apache.pekko.actor.testkit.typed.scaladsl.ScalaTestWithActorTestKit
+import org.apache.pekko.actor.typed.scaladsl.AskPattern.Askable
+import org.apache.pekko.actor.typed.scaladsl.AskPattern.schedulerFromActorSystem
+import org.apache.pekko.actor.typed.scaladsl.Behaviors
+import org.apache.pekko.actor.typed.scaladsl.Behaviors._
+import org.apache.pekko.util.Timeout
+import org.scalatest.wordspec.AnyWordSpecLike
+
+import java.util.concurrent.CountDownLatch
+import scala.concurrent.ExecutionContext
+import scala.concurrent.Future
+import scala.concurrent.TimeoutException
+import scala.concurrent.duration._
+import scala.util.Failure
+import scala.util.Success
+
+sealed trait Command
+case class Multiply(a: Int, b: Int, forwardRef: ActorRef[WorkerCommand], replyTo: ActorRef[Int]) extends Command
+case class ReplyResult(num: Int, replyTo: ActorRef[Int]) extends Command
+case class Ignore() extends Command
+
+sealed trait WorkerCommand
+case class WorkerMultiply(a: Int, b: Int, replyTo: ActorRef[WorkerResult]) extends WorkerCommand
+case class WorkerResult(num: Int) extends WorkerCommand
+
+class ManualTerminatedTestSetup(workerCnt: Int) {
+  implicit val timeout: Timeout = 10.millis
+  val workerLatch = new CountDownLatch(workerCnt)
```

Comment:
this is bad practice for unit tests, multiple tests in DeadLetterSpec may use this same immutable value.